gemrb/core/Audio/AmbientMgr.cpp

```diff
@@ -309,12 +302,37 @@ tick_t AmbientMgr::AmbientSource::Tick(tick_t ticks, Point listener, ieDword tim
 	return nextdelay;
 }
 
+BufferCacheEntry AmbientMgr::AmbientSource::GetBuffer(ResRef resource, const AudioPlaybackConfig& config)
+{
+	auto entry = bufferCache.get().Lookup(resource);
+	if (entry) {
+		return *entry;
+	}
+
+	ResourceHolder<SoundMgr> acm = gamedata->GetResourceHolder<SoundMgr>(resource);
+	auto length = ((acm->get_length() / acm->get_channels()) * 1000) / acm->get_samplerate();
```

Comment:
Is there any chance for this to SIGFPE?

Reply:
Not sure, I have taken this from the previous code. Will check.